lib/explorer/polars_backend/data_frame.ex

```diff
@@ -643,22 +643,44 @@ defmodule Explorer.PolarsBackend.DataFrame do
   end
 
   @impl true
-  def arrange_with(%DataFrame{} = df, out_df, column_pairs, nulls_last, maintain_order) do
-    {directions, expressions} =
-      column_pairs
-      |> Enum.map(fn {direction, lazy_series} ->
-        expr = to_expr(lazy_series)
-        {direction == :desc, expr}
-      end)
-      |> Enum.unzip()
-
-    Shared.apply_dataframe(df, out_df, :df_arrange_with, [
-      expressions,
-      directions,
-      nulls_last,
-      maintain_order,
-      df.groups
-    ])
+  def arrange_with(
+        %DataFrame{} = df,
+        out_df,
+        column_pairs,
+        maintain_order?,
+        multithreaded?,
+        nulls_last?
+      )
+      when is_boolean(maintain_order?) and is_boolean(multithreaded?) and
+             is_boolean(nulls_last?) do
+    if Enum.all?(column_pairs, fn {_, %{op: op}} -> op == :column end) do
+      {directions, column_names} =
+        column_pairs
+        |> Enum.map(fn {dir, %{args: [col]}} -> {dir == :desc, col} end)
+        |> Enum.unzip()
+
+      Shared.apply_dataframe(df, out_df, :df_arrange, [
+        column_names,
+        directions,
+        maintain_order?,
+        multithreaded?,
+        nulls_last?,
+        df.groups
+      ])
```

Comment:
We weren't ever actually calling `df_arrange` from the Elixir side. You can think of this clause as basically an optimization.